simulator/crypto_manager_kmip.go

```diff
@@ -0,0 +1,515 @@
+/*
+Copyright (c) 2024-2024 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package simulator
+
+import (
+	"slices"
+
+	"github.com/google/uuid"
+
+	"github.com/vmware/govmomi/vim25/methods"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	nativeKeyProvider = string(types.KmipClusterInfoKmsManagementTypeNativeProvider)
+)
+
+type CryptoManagerKmip struct {
+	mo.CryptoManagerKmip
+
+	keyIDToProviderID map[string]string
+}
+
+func (m *CryptoManagerKmip) init(r *Registry) {
+	if m.keyIDToProviderID == nil {
+		m.keyIDToProviderID = map[string]string{}
+	}
+}
+
+func (m CryptoManagerKmip) moRefEq(a, b types.ManagedObjectReference) bool {
+	return a.Type == b.Type && a.Value == b.Value
```

Comment:
Does `a == b` not work? We use that quite a bit elsewhere. Or is it related to ServerGUID?